### PR TITLE
Update documentation and configs for VSCode Marketplace

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

To prep the extension for the VSCode marketplace

VSCode guidelines for publishing to marketplace
https://code.visualstudio.com/api/working-with-extensions/publishing-extension
https://code.visualstudio.com/api/references/extension-manifest#marketplace-presentation-tips
 

### WHAT is this pull request doing?
* updating the readme to be displayed on the VSCode Marketplace
* Adding a .gif demo of the extension in action
* Updating Changelog, contributing, and license files

### CODE REVIEW

File: polaris-for-vscode/README.md [hunk 1]
```diff
@@ -1,38 +1,26 @@
-# `polaris-for-vscode` Extension
+# Polaris for VSCode
 
-:warning: This extension is still an experimental prototype
+Official VSCode extension for building with the Shopify [Polaris Design System](https://polaris.shopify.com/).
 
-VSCode features to support developing with the Polaris Design System.
+![Demo of Polaris for figma autocompleting token code](https://github.com/Shopify/polaris/blob/main/polaris-for-vscode/docs/polaris-for-vscode-preview.gif?raw=true)
```
Comment (yurm04) [inline polaris-for-vscode/README.md:5]: I can't get the extension marketplace to resolve a relative URL so we will need to merge in this PR first before it shows up correctly in the packaged extension

Comment (yurm04) [reply polaris-for-vscode/README.md:5]: If you check out the [shopify theme check](https://github.com/Shopify/theme-check-vscode/blob/main/README.md?plain=1#L3) extension they also had to use this method

File: polaris-for-vscode/CHANGELOG.md [hunk 2]
```diff
@@ -6,4 +6,5 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 
 ## [Unreleased]
 
+## [1.0.0] 2022-04-12
```
Comment (alex-page) [inline polaris-for-vscode/CHANGELOG.md]: Should we assume this is `v0.1.0` as it's still being iterated upon?

File: polaris-for-vscode/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2017-present Shopify Inc.
```
Comment (alex-page) [inline polaris-for-vscode/LICENSE.md]: Can we just have on LICENSE in the root of the monorepo? Does this get included in the published files?

File: polaris-for-vscode/server/package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "polaris-for-vscode",
   "description": "Polaris Design System VSCode extension",
-  "version": "0.0.1",
+  "version": "1.0.0",
```
Comment (alex-page) [inline polaris-for-vscode/server/package.json]: Question about version above ^

**Review by alex-page — APPROVED** [review]

**Review by romellogoodman — APPROVED** [review]

**Review by rahilllshah — APPROVED** [review]

**Pull request merged**